web/server/entrypoint_test.go

```diff
@@ -522,3 +522,112 @@ func TestTunnelE2E(t *testing.T) {
+	cfg := &config.Config{
+		Modules: []config.Module{
+			{
+				Name:    "testModule",
+				ExePath: testModulePath,
+			},
+		},
+		Components: []resource.Config{
+			{
+				Name:  "helper",
+				API:   generic.API,
+				Model: helperModel,
+			},
+		},
+		Network: config.NetworkConfig{
+			NetworkConfigData: config.NetworkConfigData{
+				BindAddress: machineAddress,
+			},
+		},
+	}
+	cfgBytes, err := json.Marshal(&cfg)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, os.WriteFile(tempConfigFile.Name(), cfgBytes, 0o755), test.ShouldBeNil)
+
+	// Call `RunServer` in a goroutine as it is blocking. Point it to the temporary config
+	// file created above.
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		args := []string{"viam-server", "-config", tempConfigFile.Name()}
+		test.That(t, server.RunServer(ctx, args, logger), test.ShouldBeNil)
+	}()
+
+	// Create an SDK client to the server that was started on localhost:23654.
+	rc := robottestutils.NewRobotClient(t, logger, machineAddress, time.Second)
+	helper, err := rc.ResourceByName(generic.Named("helper"))
+	test.That(t, err, test.ShouldBeNil)
+
+	// Log a DEBUG line through helper. While we cannot actually examine the log output, we
+	// can examine the response from the component to see its set log level. That level
+	// should start as "Info."
+	resp, err := helper.DoCommand(ctx,
+		map[string]any{"command": "log", "msg": "debug log line", "level": "DEBUG"})
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, resp, test.ShouldResemble, map[string]any{"level": "Info"})
+
+	// Write an identical config to the temporary config file but add { "debug": true }.
+	cfg.Debug = true
+	cfgBytes, err = json.Marshal(&cfg)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, os.WriteFile(tempConfigFile.Name(), cfgBytes, 0o755), test.ShouldBeNil)
+
+	// Wait for the helper to reconfigure and report a log level of "Debug."
+	gtestutils.WaitForAssertion(t, func(tb testing.TB) {
+		resp, err = helper.DoCommand(ctx,
+			map[string]any{"command": "log", "msg": "debug log line", "level": "DEBUG"})
+		test.That(tb, err, test.ShouldBeNil)
+		test.That(tb, resp, test.ShouldResemble, map[string]any{"level": "Debug"})
+	})
+
+	// Change "debug" to be false in the temporary config file.
+	cfg.Debug = false
+	cfgBytes, err = json.Marshal(&cfg)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, os.WriteFile(tempConfigFile.Name(), cfgBytes, 0o755), test.ShouldBeNil)
+
+	// Wait for the helper to reconfigure and report a log level of "Info."
+	gtestutils.WaitForAssertion(t, func(tb testing.TB) {
+		resp, err = helper.DoCommand(ctx,
+			map[string]any{"command": "log", "msg": "debug log line", "level": "DEBUG"})
+		test.That(tb, err, test.ShouldBeNil)
+		test.That(tb, resp, test.ShouldResemble, map[string]any{"level": "Info"})
+	})
+
+	// Specify a "log" pattern of { "testModule": "debug" } in the temporary config file.
+	cfg.LogConfig = []logging.LoggerPatternConfig{{"testModule", "debug"}}
```

Comment:
can we assert that removing the log pattern resets the log level?

Reply:
Done.